frontend/src/views/Announcements/AnnouncementsList.tsx

```diff
@@ -0,0 +1,213 @@
+import { GeneralAPIError } from '../../components/GeneralAPIError/GeneralAPIError';
+import { useActionModal } from '../../hooks/useActionModal/useActionModal';
+import { AnnouncementCard } from './AnnouncementCard/AnnouncementCard';
+import { useAuth } from '../../hooks/useAuth/useAuth';
+import { ActionModal } from '../../components/ActionModal/ActionModal';
+import { useCommonSnackbars } from '../../hooks/useCommonSnackbars/useCommonSnackbars';
+import { useMakeRequestWithAuth } from '../../hooks/useMakeRequestWithAuth/useMakeRequestWithAuth';
+import { AnnouncementFormModal } from './AnnouncementFormModal/AnnouncementFormModal';
+import { AnnouncementFormSchema } from './AnnouncementFormModal/AnnouncementFormModal.schema';
+
+export const AnnouncementsList = () => {
+  const {
+    data: announcementsData,
+    error: announcementsError,
+    mutate,
+  } = useSWR<AnnouncementFindAllResponseDto>('/api/announcements');
+  const pageTitle = 'Ogłoszenia';
+  const { role } = useAuth();
+  const { showErrorSnackbar, showSuccessSnackbar } = useCommonSnackbars();
+  const makeRequest = useMakeRequestWithAuth();
+  const {
+    setLoading: setDeleteModalLoading,
+    isLoading: isDeleteModalLoading,
+    isOpen: isDeleteModalOpen,
+    openModal: openDeleteModal,
+    closeModal: closeDeleteModal,
+  } = useActionModal();
+
+  const [announcementToBeDeleted, setAnnouncementToBeDeleted] =
+    useState<number>();
+  const [announcementToBeEdited, setAnnouncementToBeEdited] =
+    useState<DtoAnnouncement>();
+  const [isCreateAnnouncementModalOpen, setIsCreateAnnouncementModalOpen] =
+    useState(false);
+  const [isEditAnnouncementModalOpen, setIsEditAnnouncementModalOpen] =
+    useState(false);
+
+  const handleDelete = (id: number) => {
+    openDeleteModal();
+    setAnnouncementToBeDeleted(id);
+  };
+
+  const deleteAnnouncement = async () => {
+    setDeleteModalLoading(true);
+    const announcementsApiUrl = `/api/announcements/${announcementToBeDeleted}`;
+    const response = await makeRequest<AnnouncementDeleteResponseDto>(
+      announcementsApiUrl,
+      'DELETE',
+    );
+    setDeleteModalLoading(false);
+    if (!response.ok) {
+      showErrorSnackbar();
+      return;
+    }
+    setDeleteModalLoading(false);
+    closeDeleteModal();
+    await mutate();
+    showSuccessSnackbar('Ogłoszenie zostało pomyślnie usunięte');
+  };
+
+  const handleCreate = () => {
+    setIsCreateAnnouncementModalOpen(true);
+  };
+
+  const createAnnouncement = async (values: AnnouncementFormSchema) => {
+    const { subject, body } = values;
+    const announcement: AnnouncementCreateRequestDto = {
+      announcement: {
+        subject,
+        body,
+      },
+    };
+    const announcementApiUrl = '/api/announcements';
+    const response = await makeRequest<
+      AnnouncementCreateResponseDto,
+      AnnouncementCreateRequestDto
+    >(announcementApiUrl, 'POST', announcement);
+    if (!response.ok) {
+      showErrorSnackbar();
```

Comment:
Probably we should make a return after this. Right now if you have an error, you'll get the "Failed creation successfully" situation 😉